testdata/sqllogictests/catalog/functions.slt

```diff
@@ -55,4 +54,35 @@ from glare_catalog.functions
 where function_name = 'read_parquet';
 ----
 read_parquet   table   Utf8/List<Utf8>   t   SELECT * FROM read_parquet('./my_data.parquet')   Returns a table by scanning the given Parquet file(s).
-read_parquet   table   Utf8/List<Utf8>   t   SELECT * FROM read_parquet('./my_data.parquet')   Returns a table by scanning the given Parquet file(s).
+
+# Assert an arbitrary glaredb table function exists (using an alias).
+query TTTTTT
+select
+    function_name,
+    function_type,
+    parameters,
+    builtin,
+    example,
+    description
+from glare_catalog.functions
+where function_name = 'parquet_scan';
+----
+parquet_scan   table   Utf8/List<Utf8>   t   SELECT * FROM read_parquet('./my_data.parquet')   Returns a table by scanning the given Parquet file(s).
```

Comment:
so i didn't have a chance to comment on it in the previous [pr](https://github.com/GlareDB/glaredb/pull/2367), but to me it seems wrong that the function name is 'parquet_scan' and the example uses 'read_parquet'.

Reply:
I agree with this, but I didn't feel like getting examples exactly correct was the biggest priority since the previous version was incorrect here too (on prod):
<img width="1123" alt="Screenshot 2024-01-08 at 1 31 49 PM" src="https://github.com/GlareDB/glaredb/assets/4040560/5bf714a4-6486-422d-8acc-b290a86153ae">